doc/cephadm/concepts.rst

```diff
@@ -0,0 +1,72 @@
+Cephadm Concepts
+================
+
+.. _cephadm-fqdn:
+
+Fully qualified domain names vs bare host names
+-----------------------------------------------
+
+cephadm has very minimal requirements when it comes to resolving host
+names etc. When cephamd initiates an ssh connection to a remote host,
+the host name  can be resolved in four different ways:
+
+-  a custom ssh config resolving the name to an IP
+-  via an externally maintained ``/etc/hosts``
+-  via explictly providing an IP address to cephadm: ``ceph orch host add <hostname> <IP>``
+-  automatic name resolution via DNS.
+
+Ceph itself uses the command ``hostname`` to determine the name of the
+current host.
+
+.. note::
+
+  cephadm demands that the name of the host given via ``ceph orch host add`` 
+  equals the output of ``hostname`` on remote hosts.
+
+Otherwise cephadm can't be sure, the host names returned by
+``ceph * metadata`` match the hosts known to cephadm. This might result  
+in a :ref:`cephadm-stry-host` warning.
+
+When configuring new hosts, there are two **valid** ways to set the 
+``hostname`` of a host:
+
+1. Using the bare host name. In this case:
+
+-  ``hostname`` returns the bare host name.
+- ``hostname -f`` returns the FQDN.
+
+2. Using the fully qualified domain name as the host name. In this case:
+
+-  ``hostname`` returns the FQDN
+-  ``hostname -s`` return the bare host name
+
+Note that ``man hostname`` recomments ``hostname`` to return the bare
+host name:
+
+    THE FQDN The FQDN (Fully Qualified Domain Name) of the system is the
+    name that the resolver(3) returns for the host name, such as,
+    ursula.example.com. It is usually the hostname followed by the DNS
+    domain name (the part after the first dot). You can check the FQDN
+    using hostname --fqdn or the domain name using dnsdomainname.
+
+    ::
+
+          You cannot change the FQDN with hostname or dnsdomainname.
+
+          The recommended method of setting the FQDN is to make the hostname
+          be an alias for the fully qualified name using /etc/hosts, DNS, or
+          NIS. For example, if the hostname was "ursula", one might have
+          a line in /etc/hosts which reads
+
+                 127.0.1.1    ursula.example.com ursula
+
+Which means, ``man hostname`` recommends ``hostname`` to return the bare
+host name. This in turn means that Ceph will return the bare host names
+when executing ``ceph * metadata``. This in turn means cephadm also
+requires the bare host name when adding a host to the cluster: 
+``ceph orch host add <bare-name>``.
+
+::
+
+  TODO: This chapter needs to provide way for users to configure
+  Grafana in the dashboard, as this is right no very hard to do.
```

Comment:
this sentence doesn't sound right.    maybe this is better    "as this is right no very hard to do" -> "as this is very hard to do right now"

Reply:
Since the "TODO" is a note from Sebastian to Sebastian himself and no user will see it, this pull request is okay.

There are some minor changes to the English text that I would make if I were writing it, but those minor changes are not more important than approving this PR in its current state, which is sufficient to inform Ceph users of aspects of FQDN and Ceph.

I am therefore approving this pull request.